web/src/server/auth.ts

```diff
@@ -139,6 +140,13 @@ if (env.SMTP_CONNECTION_URL && env.EMAIL_FROM_ADDRESS) {
   );
 }
 
+// Function to be used in the prismaAdapter to cleanup extra
+// fields in the Account Object of specific Providers. Default does
+// nothing.
+let cleanupAccount = (account: AdapterAccount) => {
```

Comment:
I think adding a mutable function that is updated when keycloak is used can can have some side effects on other providers on the same langfuse instance.

Is this strictly necessary? It seems like this issue was addressed here:
- https://github.com/nextauthjs/next-auth/pull/7369
- https://github.com/nextauthjs/next-auth/issues/7655

If not, is there a cleaner way of only using this function on the KeycloakProvider?

Reply:
Thanks a lot for the links! I'll check them out and the code from next-auth. There probably is a better way because we surely are not the first to try to use Keycloak with next-auth!

Reply:
Did I understood it correctly that this change will happen only in next-auth v5? This [message](https://github.com/nextauthjs/next-auth/issues/7655#issuecomment-1567440434) seems to imply those changes were merged to main but will only be released with v5 and that until then such a "workaround" is fine (scary quotes in original). Though I suspect your concern is valid in case of a Langfuse Server being integrated with multiple IPs at the same time.

The [docs](https://auth-docs-git-feat-nextjs-auth-authjs.vercel.app/guides/upgrade-to-v5) also mention it as a breaking change when upgrading to v5 (see last bullet point in Breaking Changes).

Reply:
@mparada-suva Yeah, I think so. (https://github.com/langfuse/langfuse/pull/2866) My solution is a workaround by removing the data before linking accounts, but as a contributor to this issue, https://github.com/nextauthjs/next-auth/issues/7655 mention that this issue should be solved in version v5 by letting developers manually pass chosen fields in the provider's account callback. So, if Langfuse's team has a plan to migrate this library, they may need to make an adjustment to support this change.

Reply:
@mparada-suva  BTW, if you have any suggestions for my PR, feel free to add them. I feel my solution is quite bulky since each IP have own additional parameters, so we may need to add a code to delete these parameters in each IP, like this [comment](https://github.com/orgs/langfuse/discussions/2165#discussioncomment-10274924)

Reply:
Closing this PR in favor of #2866